l5kit/l5kit/cle/metric_set.py

```diff
@@ -0,0 +1,86 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from l5kit.cle import composite_metrics, metrics, validators
+from l5kit.cle.closed_loop_evaluator import ClosedLoopEvaluator, EvaluationPlan
+from l5kit.simulation.unroll import SimulationOutputCLE
+
+
+class BaseMetricSet(ABC):
+    """Base class interface for the metric sets."""
+    #: The prefix that will identify this metric
+    metric_prefix: str
+    evaluator: ClosedLoopEvaluator
+
+    @abstractmethod
+    def evaluate(self, sim_outputs: List[SimulationOutputCLE]) -> None:
+        """Run the CLE (Closed-loop Evaluator) on simulated scenes.
+
+        :param sim_outputs: outputs from the simulator.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_results(self) -> Dict[str, Any]:
+        """Perform all required aggregations and returns a dictionary
+        composed by all results."""
+        raise NotImplementedError
+
+
+class L5MetricSet(BaseMetricSet):
```

Comment:
shouldn't this be doing something in the methods?

Reply:
I did not understand. Should I shift L5MetricSet class to the 'environment' folder?

Reply:
what is the purpose of this class?

Reply:
Its similar in spirit to the driveformer metric_set. [Ref](https://github.com/lyft/avjadoo/blob/master/experiments/planning/driveformer_cle_metricset.py)
Its a metric set for L5Kit. 

We can have different metric sets inheriting from this. Currently, we have one with distance and yaw error. In future, we can have one with collision metrics as well.

Reply:
got it.

It's a little bit weird because this class does nothing so we could also move everything in the super. But I guess we need this hierarchy for inheritance at some point